apps/glusterfs/operations.go

```diff
@@ -671,6 +671,137 @@ func (vdel *BlockVolumeDeleteOperation) Finalize() error {
 	})
 }
 
+// DeviceRemoveOperation is a phony-ish operation that exists
+// primarily to a) know that set state was being performed
+// and b) to serve as a starting point for a more proper
+// operation in the future.
+type DeviceRemoveOperation struct {
+	OperationManager
+	DeviceId  string
+	allocator Allocator
+}
+
+// Note: passing this allocator here a big hack, but its a temporary
+// workaround for how brick relocation currently works
```

Comment:
To be even more clear, if I got it right, the hack is not only passing the allocator in to the constructor, but storing it in the operation and using it later in the exec part instead of the build part:

Usually, the allocator is passed into the `Build` method, and the `Exec` part has nothing to do with the allocator. In the case of the `DeviceRemove`, the actual code is not (yet?) untangled enough to allow for this separation. Hence the woraround.

Reply:
Correct.